File: CourseProjects/CourseProjects/VisualStudioProjects/Agile/MockingWithMoq/BasicMocking/Shop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BasicMocking {
    public class Shop {
        public Shop(IPricingEngine pricingEngine,
                    IStockCheckEngine stockCheckEngine,
                    IPaymentEngine paymentEngine) {
            this.pricingEngine = pricingEngine;
            this.stockCheckEngine = stockCheckEngine;
            this.paymentEngine = paymentEngine;
        }
        public bool makePurchase(String itemNo, int quantity, String cardNo) {
            if (stockCheckEngine.Check(itemNo) >= quantity) {
                double charge = pricingEngine.Price(itemNo, quantity);
                if (paymentEngine.Authorize(cardNo, charge)) {
                    return true;
                }
            }
            return false;
        }
        private IPricingEngine pricingEngine;
        private IStockCheckEngine stockCheckEngine;
        private IPaymentEngine paymentEngine;
    }
 
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/UnitTestingExamples/src/demos/tdd/mockito/basic/Shop.java
package demos.tdd.mockito.basic;

public class Shop {
	public Shop(PricingEngine pricingEngine, StockCheckEngine stockCheckEngine, PaymentEngine paymentEngine) {
		super();
		this.pricingEngine = pricingEngine;
		this.stockCheckEngine = stockCheckEngine;
		this.paymentEngine = paymentEngine;
	}
	public boolean makePurchase(String itemNo, int quantity, String cardNo) {
		if(stockCheckEngine.check(itemNo) >= quantity) {
			double charge = pricingEngine.price(itemNo, quantity);
			if(paymentEngine.authorize(cardNo, charge)) {
				return true;
			}
		}
		return false;
	}
    private PricingEngine pricingEngine;
    private StockCheckEngine stockCheckEngine;
    private PaymentEngine paymentEngine;
}
